src/ssd/ssd_titlebar.c

```diff
@@ -204,9 +204,10 @@ ssd_titlebar_update(struct ssd *ssd)
 		wl_list_for_each(part, &subtree->parts, link) {
 			switch (part->type) {
 			case LAB_SSD_PART_TITLEBAR:
+				/* LAB_MIN_VIEW_WIDTH < SSD_BUTTON_WIDTH * SSD_BUTTON_COUNT */
```

Comment:
Hm.. I wonder how that can happen.

Reply:
They're defined like this:
```c
#define LAB_MIN_VIEW_WIDTH  100
#define SSD_BUTTON_COUNT 4
#define SSD_BUTTON_WIDTH 26
```
Feel free to close this PR if another solution like changing these values is better.
I'm not familiar with `ssd*.c` and just blindly fixed the error from pixman.

Reply:
Whoops, that seems pretty wrong then.

Personal view: what about us doing `#define LAB_MIN_VIEW_WIDTH (SSD_BUTTON_WIDTH * SSD_BUTTON_COUNT)` but also detect the ssd_titlebar case and log an error before fixing it?

I am not quite sure currently if a view really can never be smaller than `LAB_MIN_VIEW_HEIGHT`.

Reply:
I changed it to `#define LAB_MIN_VIEW_WIDTH (SSD_BUTTON_WIDTH * SSD_BUTTON_COUNT)`.
Also, I updated `ssd_update_geometry()` to log error and early-return if the view width is smaller than that, since the same pixman error also happens in the following part in `ssd_border_update()` if the view width is very small.
```c
			case LAB_SSD_PART_TOP:
				if (ssd->titlebar.height > 0) {
					wlr_scene_rect_set_size(rect,
						width - 2 * SSD_BUTTON_WIDTH,
						theme->border_width);
``` 
> I am not quite sure currently if a view really can never be smaller than LAB_MIN_VIEW_WIDTH.

Yeah, I confirmed clients can ignore the requested geometry and `view->current.width` can be smaller than `LAB_MIN_VIEW_WIDTH` ([source](https://github.com/tokyo4j/wayland-demo/tree/deco-width-10)).
![20240427_16h37m33s_grim](https://github.com/labwc/labwc/assets/22029524/fb12e163-919d-4748-b402-5afb5eed43b7)

Reply:
> I changed it to `#define LAB_MIN_VIEW_WIDTH (SSD_BUTTON_WIDTH * SSD_BUTTON_COUNT)`.
> Also, I updated `ssd_update_geometry()` to log error and early-return if the view width is smaller than that, since the same pixman error also happens in the following part in `ssd_border_update()` if the view width is very small.

Thanks, I'll look into the diff in a few hours.

> So I suggest enforcing the minimal geometry in `view_impl_apply_geometry()`.

There must be some discussion around this somewhere, I thought we actually did that already but apparently we did not. It should also only be applied when there is actually a SSD with titlebar shown, otherwise there is not really a point in preventing smaller windows. This is something that should likely wait until after the release `0.7.2` though.